### PR TITLE
PREAPPS-3053: Login and Logout using SAML

### PR DESCRIPTION
**Pull request by panditsavitags**

### CODE REVIEW

File: src/batch-client/index.ts [hunk 1]
```diff
@@ -538,6 +539,22 @@ export class ZimbraBatchClient {
...
+		this.jsonRequest({
+			name: 'ClientInfo',
+			singleRequest: true,
+			body: {
+				hostname: hostname
+			},
+			namespace: Namespace.Account
+		}).then(() => ({
+			zimbraFeatureStaySignedInDisabled: false,
+			zimbraWebClientLoginURL: 'www.google.com',
```
Comment (itsmeprashant) [inline src/batch-client/index.ts]: Add TODO note

File: src/batch-client/index.ts [hunk 2]
```diff
@@ -538,6 +539,21 @@ export class ZimbraBatchClient {
...
+		this.jsonRequest({
+			name: 'ClientInfo',
+			singleRequest: true,
+			body: {
+				hostname: hostname
+			},
+			namespace: Namespace.Account
+		}).then(() => ({
+			//TODO: waiting for backend JSON response
+			zimbraWebClientLoginURL: 'www.google.com',
```
Comment (silentsakky) [inline src/batch-client/index.ts]: use https://www.google.com

File: src/schema/schema.ts [hunk 3]
```diff
@@ -110,6 +111,8 @@ export function createZimbraSchema(
 					client.freeBusy(variables as FreeBusyOptions),
 				getContact: (_, variables) =>
 					client.getContact(variables as GetContactOptions),
+				getClientConfig: (_, variables) =>
```
Comment (silentsakky) [inline src/schema/schema.ts]: clientInfo

File: src/schema/schema.ts [hunk 4]
```diff
@@ -110,6 +111,8 @@ export function createZimbraSchema(
...
+					client.getClientConfig(variables as ClientConfigInput),
```
Comment (silentsakky) [inline src/schema/schema.ts]: `client.clientInfo(variables as ClientInfoInput)`

File: src/schema/schema.graphql [hunk 5]
```diff
@@ -2534,6 +2543,7 @@ type Query {
 	downloadAttachment(id: ID!, part: ID!): Attachment
 	discoverRights: DiscoverRights
 	freeBusy(names: [String!]!, start: Float, end: Float): [FreeBusy]
+	getClientConfig(hostname: String): ClientConfigType
```
Comment (silentsakky) [inline src/schema/schema.graphql]: `clientInfo(hostname: String): ClientInfoType`

File: src/schema/schema.graphql [hunk 6]
```diff
@@ -449,6 +449,15 @@ interface MailItem {
 	replyType: String #rt
 }
 
+type ClientInfoAttributes {
+	zimbraWebClientLoginURL: String
+	zimbraWebClientLogoutURL: String
+}
+
+type ClientInfoType {
+	_attrs: ClientInfoAttributes
```
Comment (silentsakky) [inline src/schema/schema.graphql]: map _attrs to attributes

**Pull request merged**